src/components/SingletonEditor.tsx

```diff
@@ -24,48 +31,97 @@ interface UseSingletonEditorConfig {
   onSave: (content: Content) => void;
 }
 
+interface UnlayerEditor extends EmailEditor {
+  removeEventListener: any;
+}
+
 export const SingletonDesignContext = createContext<ISingletonDesignContext>({
   hidden: true,
   setContent: () => {},
   editorState: { isLoaded: false, unlayer: undefined },
 });
 
-export const useSingletonEditor = ({
-  initialContent,
-  onSave,
-}: UseSingletonEditorConfig) => {
+export const useSingletonEditor = (
+  { initialContent, onSave }: UseSingletonEditorConfig,
+  deps: any[]
+) => {
   const { editorState, setContent } = useContext(SingletonDesignContext);
+  const hasChangesRef = useRef(false);
 
-  const saveHandler = () => {
-    if (!editorState.isLoaded) {
-      console.error("The editor is loading, can't save yet!");
-      return;
-    }
+  const saveHandler = useCallback(
+    (force?: boolean) => {
+      if (!hasChangesRef.current && !force) {
+        return;
+      }
+      if (!editorState.isLoaded) {
+        console.error("The editor is loading, can't save yet!");
+        return;
+      }
 
-    editorState.unlayer.exportHtml((htmlExport: HtmlExport) => {
-      const content = !htmlExport.design
-        ? {
-            htmlContent: htmlExport.html,
-            type: "html",
-          }
-        : {
-            design: htmlExport.design,
-            htmlContent: htmlExport.html,
-            type: "unlayer",
-          };
-
-      onSave(content as Content);
-    });
-  };
+      editorState.unlayer.exportHtml((htmlExport: HtmlExport) => {
+        const content = !htmlExport.design
+          ? {
+              htmlContent: htmlExport.html,
+              type: "html",
+            }
+          : {
+              design: htmlExport.design,
+              htmlContent: htmlExport.html,
+              type: "unlayer",
+            };
+
+        onSave(content as Content);
+      });
+    },
+    // eslint-disable-next-line
+    [editorState, ...deps]
+  );
 
   useEffect(() => {
+    hasChangesRef.current = false;
+    window.addEventListener("beforeunload", () => saveHandler());
```

Comment:
```suggestion
    window.addEventListener("beforeunload", saveHandler);
```

Ver https://github.com/FromDoppler/doppler-editors-webapp/pull/208/files#r862005176

Reply:
💪🏻